encode.go

```diff
@@ -273,7 +274,19 @@ func (e *Encoder) encodeFloat(v float64) ast.Node {
 }
 
 func (e *Encoder) encodeString(v string, column int) ast.Node {
-	if e.isJSONStyle || token.IsNeedQuoted(v) {
+	shouldQuote := false
```

Comment:
Please refactor like the following code .

```go

func (e *Encoder) isNeedQuoted(v string) bool {
  if e.isJSONStyle {
    return true
  }
  if e.useLiteralStyleIfMultiline && strings.ContainsAny(v, "\n\r")
    return true
  }
  if token.IsNeedQuoted(v) {
    return true
  }
  return false
}

func (e *Encoder) encodeString(v string, column int) ast.Node {
  if e.isNeedQuoted(v) {
    v = strconv.Quote(v)
  }
  return ast.String(token.New(v, v, e.pos(column))
}

```

Reply:
Addressed